### PR TITLE
Update 20231130

### PR DESCRIPTION
**Pull request by cehbrecht**
## Overview

This PR is an update via cruft from the latest cookiecutter template. Is also updates the minimal Python version >=3.8 and allows installation of latest pywps = 4.6.x.

### CODE REVIEW

File: .readthedocs.yml [hunk 1]
```diff
@@ -10,21 +10,24 @@ sphinx:
...
-#python:
-#  version: 3.6
+build:
+  os: ubuntu-22.04
+  tools:
+    python: "mambaforge-22.9"
 
 conda:
-  environment: environment-docs.yml
+  environment: environment.yml
```
Comment (Zeitsperre) [inline .readthedocs.yml:28]: This should stay the same, no?
```suggestion
  environment: environment-docs.yml
```

File: CONTRIBUTING.rst
Comment (Zeitsperre) [inline CONTRIBUTING.rst:1]: This file should be updated on the cookiecutter. Lots of references to old Python version and Travis CI.

File: environment-docs.yml
Comment (Zeitsperre) [inline environment-docs.yml:1]: We should probably restrict the Python version here.

File: requirements_dev.txt [hunk 2]
```diff
@@ -1,14 +1,17 @@
 pytest>=6.0
+pytest-runner
```
Comment (Zeitsperre) [inline requirements_dev.txt:2]: `pytest-runner` is obsolete. Can likely be removed from cookiecutter.

File: requirements_dev.txt [hunk 3]
```diff
@@ -1,14 +1,17 @@
...
 ipython
 pytest-notebook
 nbsphinx
 nbval>=0.9.6
 nbconvert
-sphinx>=1.7
-bumpversion
-twine
+sphinx>=1.8.5
+bump2version
```
Comment (Zeitsperre) [inline requirements_dev.txt:13]: `bump-my-version` is the new maintained fork for this project (`bump{2}version` haven't received updates in years!). I only found this out a few weeks ago!

File: requirements_dev.txt [hunk 4]
```diff
@@ -1,14 +1,17 @@
...
+Click
```
Comment (Zeitsperre) [inline requirements_dev.txt:14]: Is `click` not installed by default with `emu`? Until we move to a new packaging system, we probably still need `twine` for package uploads.

File: tests/test_emu.py
Comment (Zeitsperre) [inline tests/test_emu.py:1]: Unless we want to add some CLI tests, we can probably remove this file.

File: tox.ini
Comment (Zeitsperre) [inline tox.ini:1]: This should be updated in the cookiecutter (remove `travis`, add more python versions envlist, use `tox` v4.0, etc.)

File: .readthedocs.yml [hunk 5]
```diff
@@ -10,21 +10,24 @@ sphinx:
   configuration: docs/source/conf.py
   # fail_on_warning might generate hard to fix error, in this case it can be
   # disabled but this also means those errors will fail silently, choose wisely.
-  fail_on_warning: true
+  fail_on_warning: false
```
Comment (Zeitsperre) [inline .readthedocs.yml:13]: I think we want this to be `true`

**Review by Zeitsperre — COMMENTED** [review]
A few changes needed here. I've been updating the cookiecutter at https://github.com/Ouranosinc/cookiecutter-pypackage a lot in the past few weeks. There are probably a good number of things we can port to the birdhouse flavour of it.

File: .readthedocs.yml (hunk 5 above)
Comment (cehbrecht) [reply .readthedocs.yml:13]: done

File: requirements_dev.txt (hunk 2 above)
Comment (cehbrecht) [reply requirements_dev.txt:2]: removed

File: CONTRIBUTING.rst
Comment (cehbrecht) [reply CONTRIBUTING.rst:1]: removed ... can be updated in the cookiecutter

File: environment-docs.yml
Comment (cehbrecht) [reply environment-docs.yml:1]: done

**Review by Zeitsperre — APPROVED** [review]

**Pull request merged**